src/core/services/interimElement/interimElement.js

```diff
@@ -323,30 +329,40 @@ function InterimElementProvider() {
        */
       function hide(reason, options) {
         if ( !stack.length ) return $q.when(reason);
+        if (hideQueue >= stack.length) return;
         options = options || {};
 
         if (options.closeAll) {
-          var promise = $q.all(stack.reverse().map(closeElement));
-          stack = [];
-          return promise;
+          return $q.all(stack.reverse().map(closeElement));
         } else if (options.closeTo !== undefined) {
-          return $q.all(stack.splice(options.closeTo).map(closeElement));
+          return $q.all(stack.slice(options.closeTo).map(closeElement));
         } else {
-          var interim = stack.pop();
-          return closeElement(interim);
+          return closeElement(stack[hideQueue]);
         }
 
         function closeElement(interim) {
+          hideQueue++;
           interim
             .remove(reason, false, options || { })
             .catch(function( reason ) {
               //$log.error("InterimElement.hide() error: " + reason );
               return reason;
+            })
+            .finally(function() {
+              handleElementHide(interim);
             });
           return interim.deferred.promise;
         }
       }
 
+      function handleElementHide(element) {
+        var index = stack.indexOf(element);
+        if (index != -1) {
+          stack.splice(index, 1);
```

Comment:
@EladBezalel - Here the items will be removed from the stack, after the animation is done.